SF iOS/SF iOS/Feed/EventsFeedViewController.m

```diff
@@ -82,12 +92,56 @@ - (void)viewDidLoad {
     [self.tableView.widthAnchor constraintEqualToAnchor:self.view.widthAnchor].active = true;
     
     self.tableView.refreshControl = [[UIRefreshControl alloc] init];
-    [self.tableView.refreshControl addTarget:self.dataSource action:@selector(refresh) forControlEvents:UIControlEventValueChanged];
+    [self.tableView.refreshControl addTarget:self action:@selector(handleRefresh) forControlEvents:UIControlEventValueChanged];
     
     if (self.traitCollection.forceTouchCapability == UIForceTouchCapabilityAvailable) {
         [self registerForPreviewingWithDelegate:self sourceView:self.tableView];
     }
     
+    CGRect searchBarRect = CGRectMake(0, 0, self.view.frame.size.width, kSEARCHBARHEIGHT);
+    self.searchBar = [[UISearchBar alloc] initWithFrame:searchBarRect];
+    self.searchBar.searchBarStyle = UISearchBarStyleMinimal;
+    self.searchBar.placeholder = NSLocalizedString(@"Filter", @"Prompt to search for event names. Here, `Filter` is a joke in English because people filter coffee and this list can be filtered by a term.");
+    self.searchBar.showsCancelButton = true;
+    self.searchBar.delegate = self;
+    
+    CGRect tableSearchViewRect = CGRectMake(0, 0, self.searchBar.frame.size.width, kTABLEHEADERHEIGHT);
+    UIView *tableSearchView = [[UIView alloc] initWithFrame:tableSearchViewRect];
+    tableSearchView.backgroundColor = UIColor.whiteColor;
+    [tableSearchView addSubview:self.searchBar];
+    self.tableView.tableHeaderView = tableSearchView;
```

Comment:
do we need to `tableSearchView`? i thing we can set `self.tableView.tableHeaderView = self.searchBar` and everything should Just Work™

(Also, UIKit will size everything for us if we call `sizeToFit` on a search bar, so we can give the searchBar a `CGSizeZero` frame and maybe not have to hardcode `kSEARCHBARHEIGHT ` or `kTABLEHEADERHEIGHT`?)